### PR TITLE
Adds Notice Component, Refactor for Alert Component

### PR DESCRIPTION
**Pull request by macandcheese**
breaking -
- `calcite-alert` - `dismiss` attribute updated to `auto-dismiss`
- `calcite-alert` - `duration` attribute updated to `auto-dismiss-duration`
- `calcite-alert` - `.openCalciteAlert()` method updated to `.open()`
- `calcite-alert` - `.closeCalciteAlert()` method updated to `.close()`
- `calcite-alert` no longer requires a wrapping `calcite-alerts` component

added -
- new component - `calcite-notice`
 -- addresses https://github.com/Esri/calcite-components/issues/176 in a new component @ehusaint
 -- the structure and props/attributes should match `calcite-alert`, with the addition of a `width` prop to support one of `[auto/half/full]` since it's an inline component
-- it does not have a queue, each notice manages it's own active state
- `calcite-alert` now has scale prop

updated -
- remove state tunnel dependency from `calcite-alert`
- removed wrapping `calcite-alerts` component - each alert component now manages it's own instance of the alert queue, rather than a parent <> child relationship
- general component update and refactoring
- tests and storybook for alert and notice

discussion @Esri/calcite-components -

- Would like feedback on the attribute / prop / method name updates for `auto-dismiss`, `auto-dismiss-duration`, `open()`, and `close()` as they are breaking changes and would like to just make them once, and match other component nomenclature (i.e .is the convention `open()` or `calciteThingOpen()` ?)
-- Should we generalize dismiss prop/attribute to dismiss=[auto/user] for other component's use case vs. a boolean here (ie - calcite-notice uses `dismissible` as a prop for "is able to be dismissed", that could be dismiss=user)
- The scale modifier spacing probably needs a pass when we do our larger spacing / sizing update @bstifle @julio8a

### CODE REVIEW

**Comment by paulcpederson** [comment]
As per feedback, in modal I've used `open()` and `close()`. There is no reason to prefix a method name, IMO

File: src/components/calcite-notice/readme.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# calcite-example
```
Comment (paulcpederson) [inline src/components/calcite-notice/readme.md]: `calcite-notice`

**Comment by paulcpederson** [comment]
spacing vars are a little clunky, we should probably define a global var and use that rather than setting spacing unit vars on all the components. So `--calcite-global-spacing-1` or something? This would enable a consuming app to change the density of their whole application by altering a few css vars

**Comment by macandcheese** [comment]
> `--calcite-global-spacing-1` or something?

Yeah, agree... we need to add a set of modifiers to `calcite-base` - and then go through and update all our components that have scale modifiers to use the new set of values.

**Comment by paulcpederson** [comment]
@driskull @pr3tori4n this includes breaking changes, so review now or forever hold your piece.

**Comment by driskull** [comment]
I'll take a look but the changes all sound good to me 👍

File: src/components/calcite-alert/calcite-alert.tsx [hunk 2]
```diff
@@ -36,108 +34,104 @@ import AlertInterface from "../../interfaces/AlertInterface";
...
-    : null;
+  /** Duration of autoDismiss (only used with `autoDismiss`) */
+  @Prop({ reflect: true, mutable: true }) autoDismissDuration:
+    | "fast"
+    | "medium"
+    | "slow" = this.autoDismiss ? "medium" : null;
 
   /** Color for the alert (will apply to top border and icon) */
-  @Prop({ reflect: true }) color: "blue" | "green" | "red" | "yellow" = "blue";
+  @Prop({ reflect: true, mutable: true }) color:
```
[the remark this answers is not in the document]
Comment (macandcheese) [reply src/components/calcite-alert/calcite-alert.tsx]: yeah, destructive for red buttons /toggles, warning/success/info/ for alert/notice/input statuses... could probably do a pass on all of that at some point, makes it a little more tied to intended use which is good.

[the remark this answers is not in the document]
Comment (driskull) [reply src/components/calcite-alert/calcite-alert.tsx]: Yep. maybe `category`, `intent`, `purpose` or something like that. Props that are tied to an intention are better :)

[the remark this answers is not in the document]
Comment (macandcheese) [reply src/components/calcite-alert/calcite-alert.tsx]: 👍 agreed - will add this to that nomenclature issue for "the big pass" when we do it.

**Pull request merged**